packages/lodestar/src/chain/stateCache/stateContextCache.ts

```diff
@@ -23,11 +27,13 @@ export class StateContextCache {
     this.maxStates = maxStates;
   }
 
-  get(root: ByteVector): CachedBeaconState<allForks.BeaconState> | null {
+  get(root: ByteVector, metrics?: IStateCacheMetrics): CachedBeaconState<allForks.BeaconState> | null {
+    metrics?.stateLookupsTotal.inc();
     const item = this.cache.get(toHexString(root));
     if (!item) {
       return null;
     }
+    metrics?.stateLookupHits.inc();
```

Comment:
I'm not sure how will we interpret this metrics. While the regen callers make sense, there are so many paths this get function can be called, register a miss but not actually be a problem. If you see https://github.com/ChainSafe/lodestar/pull/2852/files#diff-c09440f0657ffeff9d7ba94c89ff56b27b2f44c4eaf61b641c7ab28b8813be2aR183 there's a loop that may cause many cache miss while eventually resolving. We could start by only computing cache hit miss here but without labels.